### ISSUE TITLE
Support for kotlin

### ISSUE THREAD

**Issue by Sarthak30** [facebook/infer#985]
Hi Team,

Thanks for the awesome tool. Can we also have support for Kotlin Language as we are now building new android apps using kotlin.

Thanks in advance.

**Comment by martintrojer** [comment]
Hello there, there are no current plans for Kotlin support I'm afraid.

Contributions always welcome 😉

**Comment by mnayef95** [comment]
Any update?

**Comment by nobuyuki** [comment]
I've tried this one years ago. 

I forgot the details, so sorry if I said wrong things.

![27748027_10215823291478575_5139055035090812476_o](https://user-images.githubusercontent.com/234211/58364802-bed10480-7ef4-11e9-8941-511aca083d00.jpg)

It started working. But a bunch of future improvement required.
1) Some temp variable introduced by underlying library was appeared.
2) In some case false positive appeared which was not seen in Java version.
3) Code cleanup.

Basically, Kotlin support is implemented based on Java class file, so it's not too-hard compared to build-from-parser approach. But (even) this approach may introduce 1) & 2), so I think more investigation required.

**Comment by traw** [comment]
This is a very good tool. Any update on support for kotlin. Thanks

**Comment by nobuyuki** [comment]
I'd also like to hear from the infer team on official Kotlin's support!

**Comment by Sarthak30** [comment]
I am ready to invest time and energy on this, but I would need some help to get started.

**Comment by nobuyuki** [comment]
I think the Infer team has the Kotlin support in their in-house version, so I'm happy if they can release it.

**Comment by liweijian** [comment]
@nobuyuki Basically it's been one year for their in-house Kotlin version version, just wondering do you have any news?

**Comment by jvillard** [comment]
We don't have a Kotlin version at Facebook.

**Comment by liweijian** [comment]
@jvillard That makes me really sad, I wish I could use infer in my Kotlin project.

**Comment by lUjek-Sh** [comment]
just wondering do you have any news?

**Comment by ngorogiannis** [comment]
There is experimental support for Kotlin, enable with `--kotlin-capture`.

**Comment by SolomonSun2010** [comment]
> just wondering do you have any news?

+1 @ngorogiannis

**Comment by lokeshponnada** [comment]
How do you invoke kotlin support on a gradle module? Is it  like `infer run -- ./gradlew BUILDTASK --kotlin-capture`?my module is mostly in kotlin with few legacy files in java

**Comment by ngorogiannis** [comment]
Currently, there is no support for Kotlin capture on Gradle builds, I'm afraid.

There is command line support for invocations like `infer -- kotlinc source.kt` and also support for buck builds.
